engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicProcessManager.scala

```diff
@@ -96,22 +96,46 @@ class PeriodicProcessManager(val delegate: ProcessManager,
       })
   }
 
-  override def stop(name: ProcessName, savepointDir: Option[String], user: User): Future[SavepointResult] = {
-    service.deactivate(name).flatMap {
-      _ => delegate.stop(name, savepointDir, user)
-    }
+  override def stop(name: ProcessName, savepointDir: Option[String], user: User): Future[SavepointResult] = deactivate(name) {
+    delegate.stop(name, savepointDir, user)
+  }
+
+  override def cancel(name: ProcessName, user: User): Future[Unit] = deactivate(name) {
+    delegate.cancel(name, user)
   }
 
-  override def cancel(name: ProcessName, user: User): Future[Unit] = {
-    service.deactivate(name).flatMap {
-      _ => delegate.cancel(name, user)
+  private def deactivate[T](name: ProcessName)(callback: => Future[T]): Future[T] = {
+    def handleFailed(processState: Option[ProcessState], periodicDeploymentOpt: Option[PeriodicProcessDeployment]): Future[Unit] = {
+      import PeriodicProcessDeploymentStatus._
+      periodicDeploymentOpt.map { periodicDeployment =>
+        (processState.map(_.status), periodicDeployment.state.status) match {
+          case (Some(status), Failed | Deployed) if status.isFailed => service.markFailed(periodicDeployment, processState)
```

Comment:
I wonder, maybe we should move this logic to PeriodicProcessService and just invoke:
handleFinished(deployedProcess: PeriodicProcessDeployment, processState: Option[ProcessState])
but "without reschedule" here? Because there is also case, when periodicDeployment is finished, and we want to mark as finished?

Reply:
Thanks, I actually forgot about this case. 
Fixed and documented with test.